real_estate/models/estate_property.py

```diff
@@ -0,0 +1,234 @@
+
+    # Basic Property Details
+    name = fields.Char(
+        string="Name",
+        required=True,
+        help="Name of the property, e.g., 'Bhat House'"
+    )
+    postcode = fields.Char(
+        string="Postcode",
+        help="Postal code of the property"
+    )
+    description = fields.Text(
+        string="Description",
+        help="Detailed description of the property"
+    )
+    date_availability = fields.Date(
+        string="Available Date",
+        copy=False,
+        default=fields.Date.today,
+        help="Date when the property will be available for sale"
+    )
+    # Pricing Fields
+    expected_price = fields.Float(
+        string="Expected Price",
+        required=True,
+        help="Price at which the property is expected to be sold",
+        index="btree"
+    )
+    selling_price = fields.Float(
+        string="Selling Price",
+        readonly=True,
+        copy=False,
+        help="The price at which the property is actually sold"
+    )
+    # Property Features
+    bedrooms = fields.Integer(
+        string="No. of Bedrooms",
+        default=2,
+        help="Number of bedrooms in the property"
+    )
+    living_area = fields.Integer(
+        string="Living Area (sq.m)",
+        help="The total living area of the property in square meters"
+    )
+    facades = fields.Integer(
+        string="Facades",
+        help="Number of facades of the property"
+    )
+    garage = fields.Boolean(
+        string="Has Garage?",
+        help="Indicates whether the property has a garage"
+    )
+    garden = fields.Boolean(
+        string="Has Garden?",
+        help="Indicates whether the property has a garden"
+    )
+    garden_area = fields.Integer(
+        string="Garden Area (sq.m)",
+        help="The area of the garden in square meters"
+    )
+    garden_orientation = fields.Selection(
+        string="Garden Orientation",
+        selection=[
+            ('north', "North"),
+            ('south', "South"),
+            ('east', "East"),
+            ('west', "West")
+        ],
+        help="The orientation of the garden"
+    )
+    # Property Status
+    active = fields.Boolean(
+        string="Active",
+        default=True,
+        help="Indicates whether the property is active and available for sale"
+    )
+    state = fields.Selection(
+        string="Current State",
+        selection=[
+            ('new', "New"),
+            ('offer_received', "Offer Received"),
+            ('offer_accepted', "Offer Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled")
+        ],
+        required=True,
+        copy=False,
+        default='new',
+        help="The current state of the property"
+    )
+    # Relationships
+    property_type_id = fields.Many2one(
+        comodel_name='estate.property.type',
+        string="Property Type",
+        help="The type of the property (e.g., Apartment, House, etc.)"
+    )
+    _is_commercial = fields.Boolean(
```

Comment:
```suggestion
    is_commercial = fields.Boolean(
```
Why do we need this field?

Reply:
it will be used to check whether the current property type we selected is commercial or not it it is commercial then our function would run to give the domain for buyers having only is_comapany as True

Reply:
Field doesn't makes any sense it will unnecessarily will bloat the db. I asked for a python constraint